keps/sig-auth/2579-pod-isolation-policy/README.md

```diff
@@ -0,0 +1,951 @@
+we set all of them or only a subset?
+
+<<[/UNRESOLVED]>>
+
+#### Exemptions
+
+Policy exemptions can be statically configured. Exemptions must be explicitly enumerated, and don’t
+support indirection such as label or group selectors. Requests meeting criteria are completely by
+the admission controller (allow, audit and warn). Exemption dimensions include:
+
+- Usernames: requests from users with an exempt authenticated (or impersonated) username are ignored.
+- RuntimeClassNames: pods and [templated pods] with specifying an exempt runtime class name are ignored.
+- Namespaces: pods and [templated pods] in an exempt namespace are ignored.
+
+<<[UNRESOLVED]>>
+
+Should the `kube-system` namespace default to being exempt? Alternatively, we can just label it as
+privileged out of the box.
+
+<<[/UNRESOLVED]>>
+
+### Risks and Mitigations
+
+**Future proofing:** The policy versioning aspects of this proposal are designed to anticipate
+breaking changes to policies, either in light of new threats or new knobs added to pods. However, if
+there is a new feature that needs to be restricted but doesn't have a sensible hardcoded
+requirement, we would be put in a hard place. Hopefully the adoption of this proposal will
+discourage such fields from being added.
+
+**Scope creep:** How do we decide which fields are in-scope for policy restrictions? There are a
+number of fields that are relevant to local-DoS prevention that are somewhat ambiguous.
+
+**Ecosystem suppression:** In the discussions of PodSecurityPolicy replacements, there was a concern
+that whatever we picked would become not only an ecosystem standard best-practice, but practically a
+requirement (e.g. for compliance), and that in doing so we would prevent 3rd party policy
+controllers from innovating and getting meaningful adoption. To mitigate these concerns, we have
+tightly scoped this proposal, and hopefully struck an effective balance between configurability and
+usefulness that still leaves plenty of room for 3rd party extensions. We are also providing a
+library implementation and spec to encourage custom controller development.
+
+**Exemptions creep:** We have already gotten requests to trigger exemptions on various different
+fields. The more exemption knobs we add, the harder it becomes to comprehend the current state of
+the policy, and the more room there is for error. To prevent this, we should be very conservative
+about adding new exemption dimensions. The existing knobs were carefully chosen with specific
+extensibility usecases in mind.
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Updates
```

Comment:
I don't see a section dealing with exec and attach for cases where the policy has changed.  I'm ok with saying, "if the pod is in the namespace and the policy is later tighten, since we don't clean up, we still allow exec", but I'd like to see it explicitly called out.  I expect questions about that aspect.

Reply:
Added a section on subresources.